wn/_add.py

```diff
@@ -138,10 +140,23 @@ def _add_lmf(
         _update_lookup_tables(all_infos, cur)
 
         progress.flash(f'Reading {source!s}')
-        for lexicon, info in zip(lmf.load(source), all_infos):
+
+        if lmf_progress_handler is None:
+            lmf_progress_handler = ProgressHandler
+        progress_lmf = lmf_progress_handler(message='XML read')
+
+        total_items = sum(_sum_counts(info) for info in all_infos)
+        # Since we are going to call update on __next__ of the XML
+        # Event Iterator, we will see the tags twice, because we
+        # see the opening and closing tags. This is why, the total
+        # For the lmf progress handled is the double of the total
+        # sum.
```

Comment:
Interesting. Alternatively, if someone cares about the actual count, we could just update on the "end" events.

Reply:
Okay I swear I tried that but got the wrong counts. But maybe it was in the previous version of the code in which I wasn't considering the full count. I moved and addressed the comment, now it shows the actual count, not the hacky 2X